packages/host/app/components/operator-mode/code-editor.gts

```diff
@@ -171,12 +178,13 @@ export default class CodeEditor extends Component<Signature> {
     }
     let isJSON = this.args.file.name.endsWith('.json');
     let validJSON = isJSON && this.safeJSONParse(content);
-    // Here lies the difference in how json files and other source code files
-    // are treated during editing in the code editor
+
     if (validJSON && isSingleCardDocument(validJSON)) {
-      // writes json instance but doesn't update state of the file resource
-      // relies on message service subscription to update state
-      await this.saveFileSerializedCard.perform(validJSON);
+      // Does not perform the save using the card api service because that
+      // will perform a patch request, which will would not work in case the
+      // card instance has an indexing error. Instead, we save the validated card instance data
+      // directly to the file, similar to how we save the card source code
+      await this.saveCardJson.perform(content);
```

Comment:
why don't we just use `this.writeSrouceCodeToFile()` here? Ultimately `this.saveCardJson()` calls `this.writeSrouceCodeToFile()`, but confusingly we perform `reverseFileSeralization` on this. `reverseFileSeralization` should only be if you wish to use the PATCH card API so that you can serialize the card for wire format. since we are already looking at a file serialized card via the `card-src` accept header this is incorrect processing--this will add potentially conflicting properties into the underlying file--like `id`. There should not be an `id` property in the underlying file as the filename itself is the ID.